git_machete/options.py

```diff
@@ -1,12 +1,12 @@
 from typing import Optional, List
 
 from git_machete.exceptions import MacheteException
-from git_machete.git_operations import AnyBranchName, AnyRevision, LocalBranchShortName
+from git_machete.git_operations import AnyBranchName, AnyRevision, GitContext, LocalBranchShortName
 
 
 class CommandLineOptions:
 
-    def __init__(self) -> None:
+    def __init__(self, git: GitContext) -> None:
```

Comment:
Huh I wouldn't introduce that dependency. Traverse should use `CommandLineOptions` and `git config` as two independent inputs, and resolve their precedence on its own. Such logic shoudln't reside in `CommandLineOptions` &mdash; it should be just a dumb container for options, and nothing else

Reply:
Hmm, in order for `traverse` to receive `CommandLineOptions` and `git config` as two independent inputs the hole options providing would need some major refactoring or some new unnecessary params, here's why:
- We want to achieve this options override order:
   1️⃣  Default values - CommandLineOptions
   2️⃣  Config values - only exists for some options
   3️⃣  Passed flags -> values - may be all maybe be none
- ATM, traverse (actually all commands) receive command options that follow logic where 1️⃣ is overridden by 3️⃣ 
- Now, applying 2️⃣ in between them in `traverse` is difficult:
   - we can get default values pretty easily but traverse doesn't know which flags were actually passed (it only receives 
   updated default values). So in order to know which flags were actually passed you have 2 options (1 - add some 
   parameter that stores list of options that were passed, 2 - find diff between updated values and default values) -> 
   both options are IMO bad

That's why i resolved config keys straight in the CommandLineOptions. But i understand that it's just a container so i propose resolving default options from config here (`update_cli_options_using_config_keys(cli_opts, git)`):

![image](https://user-images.githubusercontent.com/86779115/198299004-cd80a69b-2df8-4517-b978-36912ef3eb18.png)

This solutions can also be generalised into other commands that might in the future use default values stored in config. @PawelLipski wdyt?

Reply:
Okay! let's proceed with `update_cli_options_using_config_keys` then

Reply:
For clarity: are you planning to apply this change on this PR?

Reply:
@PawelLipski Yes, i just did